taxi/tests/test_views.py

```diff
@@ -0,0 +1,174 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Car
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+CAR_URL = reverse("taxi:car-list")
+
+
+class PublicManufacturerTest(TestCase):
+    def test_login_required(self):
+        res = self.client.get(MANUFACTURER_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateManufacturerTest(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="test_user",
+            password="test123",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturers(self):
+        Manufacturer.objects.create(name="test_manufacturer")
+        Manufacturer.objects.create(name="test2_manufacturer")
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertEqual(response.status_code, 200)
+        manufacturers = Manufacturer.objects.all()
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
+
+    def test_create_manufacturer(self):
+        form_data = {"name": "New Manufacturer", "country": "USA"}
+        self.client.post(reverse("taxi:manufacturer-create"), data=form_data)
+        manufacturer = Manufacturer.objects.get(name=form_data["name"])
+        self.assertEqual(manufacturer.name, form_data["name"])
```

Comment:
In the `test_create_manufacturer` method, consider adding an assertion to check that the 'country' field is correctly set in the newly created manufacturer.